### ISSUE TITLE
Using Expo - Size matters is not functional sometimes on refresh and most times when remote debugging is turned on.

### ISSUE THREAD

**Issue by BrianBenninger** [nirsky/react-native-size-matters#65]
Hi,

Just looking for some info here. I have an app using expo setup with react native size matters and sometimes when the app refreshes it looks like a mess. Almost like size matters is not able to read the screen size of the device so everything set up with it is just broken looking. I can make this happen pretty consistently if I turn on remote debugging and use React Native Debugger.

Has anyone had similar issues? And if so does anyone know why? Or have a solution? At this point we are looking at getting rid of size matters because this would be unacceptable to happen to someone opening a production application from the app store.

Thanks,

**Comment by nirsky** [comment]
Hey @BrianBenninger,

Regarding your statement:

> Almost like size matters is not able to read the screen size

size matters is using React Native's [Dimensions](https://reactnative.dev/docs/dimensions) API, you can see the usage in the [scaling-utils.js](https://github.com/nirsky/react-native-size-matters/blob/master/lib/scaling-utils.js) file.

Stating there's an issue with expo remote debugging + size matters is basically stating there's an issue with expo and the dimensions API.  

I suggest logging the values returned from `Dimensions.get('window')` and see if the logged values make sense or are actually broken.

**Comment by BrianBenninger** [comment]
Fantastic! I will do this and report back. Thank you @nirsky

**Comment by BrianBenninger** [comment]
Seems this is exactly what is happening. I've managed to console log as you suggested and am getting back

fontScale: 1
height: 36
scale: 2
width: 810

When viewing on an iPad. So from what I'm understanding this isn't an issue with react-native-size-matters but actually an issue with the Dimensions module and possibly Expo?

Feel free to close the issue if this is the case. Thank you for your help.

**Comment by rmdashrfv** [comment]
I'm having the same issue. It comes up randomly. I don't think that this library is compatible with Expo.

**Comment by rmdashrfv** [comment]
@nirsky can we submit a PR to add to the README that this library may not work with Expo? I was using it and after getting it setup across 12 screens, the library stopped working in the middle of a demo and has not worked since. I for sure had not changed any code. I think it'd save a lot of people a lot of time to know to look out for this.

**Comment by nirsky** [comment]
Hey @rmdashrfv, I need to do some investigation before declaring this lib is not compatible with Expo.

Is this happening to you on production build or only during development/debugging?
What version of expo are you using? 
Might be a specific version bug.
@BrianBenninger - I'd appreciate it if you could share some more details as well.

**Comment by rmdashrfv** [comment]
@nirsky Fair, let me back up. So one thing I've been doing to work around it (so that I can make it to the production build) is I've been doing `git checkout` to a commit in the past that was last known to work while expo is running -- so this checkout triggers a hot reload, where the app successfully builds, then I `git checkout` back to `HEAD` or wherever I'm working, and once the next hot reload is triggered, everything works properly.

It will then work properly up until I restart expo. The next time I start up expo, say, the next day, size matters will not be working at all. I'm using version 0.4.0 of `react-native-size-matters` and version `^39.0.0` of Expo.

I'll let you know if I track it down better. I've been diffing between the branch and the headless commit, and I haven't seen anything that stands out yet.

**Comment by nirsky** [comment]
Thanks for the details, I'd appreciate it if you can pin it down to the exact change that caused the issue.  
Sounds like a really weird and unexpected bug, maybe if we can identify it better we can create an issue to expo, and it will be fixed in the next versions.
It will be a shame if this lib will end up unuseable on expo... 😞

**Comment by rmdashrfv** [comment]
It is with great embarrassment that I return to you, humbly asking you not to commit aggravated assault upon me. When I first found this library, I thought that it had its own special implementation of `Dimensions` but after reading what you said above, I realized it just uses the built-in tool.

But in the past, when I had started the project, I created the main stylesheet using `import { ScaledSheet, Dimensions } from 'react-native-size-matters'
` and the app didn't complain about it for weeks until I started building my a stack navigator with `react-navigation` and implementing those styles using the `undefined` Dimensions I was importing. The confusing part was that when I was testing Dimensions in other places, it was reporting accurately because by then I knew to get Dimensions out of `react-native`

**Comment by nirsky** [comment]
@rmdashrfv 🤦🏼 
Closing this issue.

**Comment by sipamungkas** [comment]
is there any update about this issue? it's already closed, but when I'm using react native expo ~52.0.37 (March 2025), the problem still appears. sometimes the size is small, sometimes the size is very big when I use "scale" from react-native-size-matters.
Thanks
<details>
  <summary>my package.json</summary>

```json
"@expo/vector-icons": "^14.0.2",
    "@react-native-async-storage/async-storage": "^2.1.2",
    "@react-native/metro-config": "^0.78.1",
    "@react-navigation/bottom-tabs": "^7.2.0",
    "@react-navigation/native": "^7.0.14",
    "@rneui/base": "^4.0.0-rc.7",
    "@rneui/themed": "^4.0.0-rc.8",
    "axios": "^1.8.1",
    "expo": "~52.0.37",
    "expo-blur": "~14.0.3",
    "expo-constants": "~17.0.7",
    "expo-dev-client": "~5.0.12",
    "expo-device": "~7.0.2",
    "expo-font": "~13.0.4",
    "expo-haptics": "~14.0.1",
    "expo-image": "~2.0.6",
    "expo-linking": "~7.0.5",
    "expo-localization": "~16.0.1",
    "expo-router": "~4.0.17",
    "expo-screen-orientation": "~8.0.4",
    "expo-splash-screen": "~0.29.22",
    "expo-status-bar": "~2.0.1",
    "expo-symbols": "~0.2.2",
    "expo-system-ui": "~4.0.8",
    "expo-web-browser": "~14.0.2",
    "i18next": "^23.15.2",
    "intl-pluralrules": "^2.0.1",
    "lodash": "^4.17.21",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "react-i18next": "^15.0.2",
    "react-native": "0.76.7",
    "react-native-element-dropdown": "^2.12.4",
    "react-native-gesture-handler": "~2.20.2",
    "react-native-mmkv": "^3.2.0",
    "react-native-reanimated": "~3.16.1",
    "react-native-safe-area-context": "4.12.0",
    "react-native-screens": "~4.4.0",
    "react-native-size-matters": "^0.4.2",
    "react-native-svg": "15.8.0",
    "react-native-svg-transformer": "^1.5.0",
    "react-native-web": "~0.19.13",
    "react-native-webview": "13.12.5",
    "zod": "^3.23.8",
    "zustand": "^5.0.3"
```
</summary>
</details>
